File: Frontend/src/types/budget/SavingsGoalCompletionCandidateDto.ts
// Server-computed proposal for a savings goal that could be marked completed
// when the user closes the month. `projectedAmountSaved` = `amountSaved` +
// `monthlyContribution` and is the value that qualified the goal — display it
// as the "would-reach this month" amount, never the raw `amountSaved`.
export type SavingsGoalCompletionCandidateDto = {
  id: string;
  sourceSavingsGoalId: string | null;
  name: string | null;
  targetAmount: number;
  amountSaved: number;
  monthlyContribution: number;
  projectedAmountSaved: number;
  remainingAfterContribution: number;
};

File: Backend/Application/DTO/Budget/Months/SavingsGoalCompletionCandidateDto.cs
namespace Backend.Application.DTO.Budget.Months;

// A savings goal that the close-month flow can finish off in one step.
//
// "Projected" reflects what AmountSaved would be once this month's
// contribution lands. The backend does not advance AmountSaved during
// close — the projection is purely a hint used to surface candidates
// and to gate which IDs may be submitted as completions.
public sealed record SavingsGoalCompletionCandidateDto(
    Guid Id,
    Guid? SourceSavingsGoalId,
    string? Name,
    decimal TargetAmount,
    decimal AmountSaved,
    decimal MonthlyContribution,
    decimal ProjectedAmountSaved,
    decimal RemainingAfterContribution
);
